tel2puml/find_unique_graphs/otel_ingestion/json_data_converter.py

```diff
@@ -0,0 +1,511 @@
+    :type field_config: `dict`[`str`,`Any`]
+    :param index: The index of the field config
+    :type index: `int`
+    :param full_path: The key of the flattened data
+    :type full_path: `str`
+    :param flattened_data: JSON data as a flattened dictionary
+    :type flattened_data: `Any`
+    :param result: The mapped data
+    :type result: `dict`[`str`, `Any`]
+    """
+    try:
+        flattened_data = dict(flattened_data)
+        if flattened_data[full_path]:
+            value_type = _get_value_type(field_config)
+            _add_or_append_value(
+                field_name, flattened_data[full_path], result, value_type
+            )
+    except KeyError:
+        # Check if we are dealing with a list like child_span_ids
+        count = 0
+        full_path = full_path + ":" + str(count)
+        while flattened_data.get(full_path):
+            value_type = _get_value_type(field_config)
+            if field_name not in result:
+                result[field_name] = []
+            result[field_name].append(flattened_data.get(full_path))
+            full_path = full_path.replace(f":{str(count)}", f":{str(count+1)}")
+            count += 1
+    except Exception as e:
+        print(f"An error occurred - {e}")
+
+
+def _get_value_type(field_config: dict[str, str]) -> str:
+    """Function that returns the type of the field. Eg. "string", "unix_nano"
+
+    :param field_config: The config for the field name
+    :type field_config: `dict`[`str`,`Any`]
+    :return: The field type
+    :rtype: `str`
+    """
+    return field_config["value_type"]
+
+
+def _add_or_append_value(
+    field_name: str, value: str | int, result: dict[str, str], value_type: str
+) -> None:
+    """
+    Function that adds a new value or appends to an existing one in the result
+    dictionary.
+
+    :param field_name: The field name within the mapping config
+    :type field_name: `str`
+    :param value: The value pulled from the JSON data
+    :type value: `str` | `int`
+    :param result: The mapped data
+    :type result: `dict`[`str`, `str`]
+    :param value_type: The field type
+    :return value_type: `str`
+    """
+    if value_type == "unix_nano":
+        if isinstance(value, int):
+            value = _unix_nano_to_datetime_str(value)
+    if not isinstance(value, str):
+        value = str(value)
+    if field_name not in result:
+        result[field_name] = value
+    else:
+        result[field_name] += f" {value}"
+
+
+def _unix_nano_to_datetime_str(unix_nano: int) -> str:
+    """Function to process time from unix nano to datetime string.
+
+    :param unix_nano: The time in unix nano format
+    :type unix_nano: `int`
+    :return: The time in datetime string format
+    :rtype: `str`
+    """
+    dt = datetime.fromtimestamp(unix_nano // 1000000000)
+    return dt.strftime("%Y-%m-%d %H:%M:%S")
+
+
+def flatten_and_map_data(
+    json_config: JSONDataSourceConfig, raw_json: dict[str, Any]
```

Comment:
so I think you'll need to add the header stuff here. The idea was that sometimes the header for the group of spans would set some of the fields in OTelEvent and could be used across that group of spans.